api/provider/aws/environment_list_test.go

```diff
@@ -0,0 +1,73 @@
+package aws
+
+import (
+	"testing"
+
+	"github.com/quintilesims/layer0/common/db/tag_store"
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEnvironment_listTags(t *testing.T) {
+	tagStore := tag_store.NewMemoryTagStore()
+	environment := NewEnvironmentProvider(nil, tagStore, nil)
+
+	tags := models.Tags{
+		{
+			EntityID:   "eid1",
+			EntityType: "environment",
+			Key:        "name",
+			Value:      "ename1",
+		},
+		{
+			EntityID:   "eid1",
+			EntityType: "environment",
+			Key:        "os",
+			Value:      "eos1",
+		},
+		{
+			EntityID:   "eid2",
+			EntityType: "environment",
+			Key:        "name",
+			Value:      "ename2",
+		},
+		{
+			EntityID:   "eid2",
+			EntityType: "environment",
+			Key:        "os",
+			Value:      "eos2",
+		},
+		{
+			EntityID:   "someid",
+			EntityType: "environment",
+			Key:        "name",
+			Value:      "badname",
+		},
+		{
+			EntityID:   "eid1",
+			EntityType: "service",
+			Key:        "name",
+			Value:      "servicename",
+		},
+	}
+
+	for _, tag := range tags {
+		if err := tagStore.Insert(tag); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	results := []models.EnvironmentSummary{
+		{EnvironmentID: "eid1"},
+		{EnvironmentID: "eid2"},
+	}
+
+	if err := environment.listTags(results); err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, "ename1", results[0].EnvironmentName)
```

Comment:
Would it be a good test to also check for the number of items you are expecting to return based on the items you've added to tagstore?